### PR TITLE
Improve kubelet config type documentation

### PR DESCRIPTION
**Pull request by tengqm**
**What type of PR is this?**

/kind cleanup
/kind documentation
/sig node

**What this PR does / why we need it**:

There are some validation logics not captured in the kubelet config documentation. This PR adds them to improve user experience.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by gabhi** [comment]
some checks have failed

**Comment by tengqm** [comment]
@sig-node-reviewers

**Review by ehashman — APPROVED** [review]
/lgtm

Great doc changes! These look really helpful.

**Review by ehashman — COMMENTED** [review]
/lgtm

File: staging/src/k8s.io/kubelet/config/v1beta1/types.go [hunk 1]
```diff
@@ -636,11 +657,12 @@ type KubeletConfiguration struct {
...
 	EvictionMinimumReclaim map[string]string `json:"evictionMinimumReclaim,omitempty"`
-	// podsPerCore is the maximum number of pods per core. Cannot exceed MaxPods.
-	// If 0, this field is ignored.
+	// podsPerCore is the maximum number of pods per core. Cannot exceed maxPods.
+	// The value must be a non-negative integer.
+	// If 0, there is no limit on the number of Pods.
 	// Dynamic Kubelet Config (beta): If dynamically updating this field, consider that
 	// changes may cause Pods to fail admission on Kubelet restart, and may change
-	// the value reported in Node.Status.Capacity[v1.ResourcePods], thus affecting
+	// the value reported in `Node.status.capacity.pods`, thus affecting
```
Comment (derekwaynecarr) [inline staging/src/k8s.io/kubelet/config/v1beta1/types.go:665]: node.status.capacity[`pods`]

**Review by derekwaynecarr — APPROVED** [review]
a minor nit that I don't think is worth holding out on this PR as it is an improvement.

/approve
/lgtm

**Review by ohbus — APPROVED** [review]
/lgtm

**Comment by tengqm** [comment]
Anything else I can do to help get this in?

**Pull request closed without merging**